cli/azd/pkg/tools/tool.go

```diff
@@ -31,3 +36,38 @@ func toolInPath(name string) (bool, error) {
 		return false, fmt.Errorf("failed searching for `%s` on PATH: %w", name, err)
 	}
 }
+
+type ErrSemver struct {
+	ToolName    string
+	ToolRequire ToolMetaData
+}
+
+type ToolMetaData struct {
+	MinimumVersion semver.Version
+	UpdateCommand  string
+}
+
+func (err *ErrSemver) Error() string {
+	return fmt.Sprintf("need at least version %s or later of %s installed. %s %s version",
+		err.ToolRequire.MinimumVersion.String(), err.ToolName, err.ToolRequire.UpdateCommand, err.ToolName)
+}
+
+func versionToSemver(CLIOutput []byte) (semver.Version, error) {
```

Comment:
NIT: could you rename this to `cliOutput` instead of `CLIOutput`?  Typically, method parameters are camel cased)